File: Services/OnlineShop/OnlineShop.Catalog.API/CustomTypes/FilterOptions.cs
namespace OnlineShop.Catalog.API.CustomTypes
{
    public class FilterOptions
    {
        public decimal MinPrice { get; private set; }
        public decimal MaxPrice { get; private set; }
        public bool IsAvailable { get; private set; }
        public bool IsDiscounted { get; private set; }
    }
}

File: Services/WebUI/client-app/src/global/models/filterOptions.ts
export interface FilterOptions {
  searchPhrase: string | null
  subcategory: string | null
  minPrice: number | null;
  maxPrice: number | null;
  isAvailable: boolean;
  isDiscounted: boolean;
  sortProperty: string
  sortDirection: 'asc' | 'desc';
}

export class FilterOptions implements FilterOptions {
  constructor() {
    this.searchPhrase = null
    this.subcategory = null
    this.minPrice = null
    this.maxPrice = null
    this.isAvailable = false
    this.isDiscounted = false
    this.sortProperty = 'name'
    this.sortDirection = 'asc'
  }
}